### PR TITLE
[Az.EventGrid] Add partner features from 2022-06-15 API

### PR DESCRIPTION
**Pull request by Brandon411**
## Description

Design review: https://github.com/Azure/azure-powershell-cmdlet-review-pr/issues/1252

Adding new cmdlets:
- Get-AzEventGridPartnerRegistration
- Get-AzEventGridPartnerConfiguration
- Get-AzEventGridVerifiedPartner
- Get-AzEventGridPartnerNamespace
- Get-AzEventGridChannel
- Get-AzEventGridPartnerNamespaceKey
- Get-AzEventGridPartnerTopic
- Get-AzEventGridPartnerTopicEventSubscription
- New-AzEventGridPartnerRegistration
- New-AzEventGridPartnerConfiguration
- New-AzEventGridPartnerNamespace
- New-AzEventGridPartnerTopic
- New-AzEventGridChannel
- New-AzEventGridPartnerTopicEventSubscription
- Update-AzEventGridPartnerConfiguration
- Update-AzEventGridChannel
- Update-AzEventGridPartnerTopic
- Update-AzEventGridPartnerTopicEventSubscription
- Remove-AzEventGridPartnerRegistration
- Remove-AzEventGridPartnerConfiguration
- Remove-AzEventGridPartnerNamespace
- Remove-AzEventGridChannel
- Remove-AzEventGridPartnerTopic
- Remove-AzEventGridPartnerTopicEventSubscription
- Grant-AzEventGridPartnerConfiguration
- Enable-AzEventGridPartnerTopic
- Revoke-AzEventGridPartnerConfiguration
- New-AzEventGridPartnerNamespaceKey

## Checklist

- [x] Check this box to confirm: **I have read the [_Submitting Changes_](../blob/main/CONTRIBUTING.md#submitting-changes) section of [`CONTRIBUTING.md`](../blob/main/CONTRIBUTING.md) and reviewed the following information:**

* **SHOULD** select appropriate branch. Cmdlets from Autorest.PowerShell should go to [`generation`](https://github.com/Azure/azure-powershell/tree/generation) branch. 
* **SHOULD** make the title of PR clear and informative, and in the present imperative tense. 
* **SHOULD** update `ChangeLog.md` file(s) appropriately
    * For any service, the `ChangeLog.md` file can be found at `src/{{SERVICE}}/{{SERVICE}}/ChangeLog.md`
    * A snippet outlining the change(s) made in the PR should be written under the `## Upcoming Release` header in the past tense. Add changelog in description section if PR goes into [`generation`](https://github.com/Azure/azure-powershell/tree/generation) branch.
    * Should **not** change `ChangeLog.md` if no new release is required, such as fixing test case only.
* **SHOULD** have approved design review for the changes in [this repository](https://github.com/Azure/azure-powershell-cmdlet-review-pr) ([_Microsoft internal only_](../blob/main/CONTRIBUTING.md#onboarding)) with following situations
    * Create new module from scratch
    * Create new resource types which are not easy to conform to [Azure PowerShell Design Guidelines](../blob/main/documentation/development-docs/design-guidelines)
    * Create new resource type which name doesn't use module name as prefix
    * Have design question before implementation
* **SHOULD** regenerate markdown help files if there is cmdlet API change. [Instruction](../blob/main/documentation/development-docs/help-generation.md#updating-all-markdown-files-in-a-module)
* **SHOULD** have proper test coverage for changes in pull request.
* **SHOULD NOT** introduce [breaking changes](../blob/main/documentation/breaking-changes/breaking-changes-definition.md) in Az minor release except preview version.
* **SHOULD NOT** adjust version of module manually in pull request

### CODE REVIEW

**Comment by wyunchi-ms** [comment]
Do you want to go to this release or next release? If this release, please retarget to release branch `release-2022-10-12`

File: src/EventGrid/EventGrid/Az.EventGrid.psd1 [hunk 1]
```diff
@@ -12,7 +12,7 @@
 # RootModule = ''
 
 # Version number of this module.
-ModuleVersion = '1.4.0'
+ModuleVersion = '1.5.0'
```
Comment (wyunchi-ms) [inline src/EventGrid/EventGrid/Az.EventGrid.psd1:15]: No need to update the version manually. Our tools will do it when build the release.

File: src/EventGrid/EventGrid/ChangeLog.md [hunk 2]
```diff
@@ -27,6 +27,21 @@
   * IsNullOrUndefined
   * IsNotNull
 
+## Version 1.50
```
Comment (wyunchi-ms) [inline src/EventGrid/EventGrid/ChangeLog.md:30]: 1. No need to update the version manually. Our tools will do it when build the release.
1. Just add changelog to section `Upcoming Release`

Comment (Brandon411) [reply src/EventGrid/EventGrid/ChangeLog.md]: Removed manual version increment and added changelog to the `Upcoming Release` section

File: src/EventGrid/EventGrid/Az.EventGrid.psd1 (hunk 1 above)
Comment (Brandon411) [reply src/EventGrid/EventGrid/Az.EventGrid.psd1]: Removed manual version increment

**Comment by Brandon411** [comment]
> 

When is the next release? I don't think we will make the 10/12/2022 date.

**Comment by VeryEarly** [comment]
> > Do you want to go to this release or next release? If this release, please retarget to release branch `release-2022-10-12`
> 
> When is the next release? I don't think we will make the 10/12/2022 date.

it will be 2022-11-01

**Comment by Brandon411** [comment]
> please fix examples: https://dev.azure.com/azclitools/public/_build/results?buildId=8328&view=logs&jobId=e1b7b984-9f58-529f-7c5a-f15f8e35cfa6&j=e1b7b984-9f58-529f-7c5a-f15f8e35cfa6&t=e8689ae8-b216-55f7-b207-4831f84ff98a
> 
> and tests: https://dev.azure.com/azclitools/public/_build/results?buildId=8328&view=logs&jobId=e1b7b984-9f58-529f-7c5a-f15f8e35cfa6&j=b17395f6-68a3-5682-0476-d3f6f1043109&t=67f633ab-4360-5839-af42-f446e0c847f0

Fixed

**Review by VeryEarly — APPROVED** [review]

**Review by VeryEarly — APPROVED** [review]

**Pull request merged**